Add EnumSource.MATCH_NONE mode (#2768)
junit-team/junit5: documentation/src/docs/asciidoc/release-notes/release-notes-5.9.0-M1.adoc

* Add support for `IterationSelector` to select a subset of the invocations of
  parameterized or dynamic tests.

> * Add support for `MATCH_NONE` mode in `EnumSource` which selects only those enum constants
>   whose names don't match any of the pattern supplied.

Reviewer: ```suggestion
* Add support for `MATCH_NONE` mode in `EnumSource` that selects only those enum constants
  whose names match none of the supplied patterns.

```
